src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,121 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface PropsFilter {
+  todosInitial: Todo[];
+  todosToRender(filteredTodos: Todo[]): void;
+}
+
+export const TodoFilter: React.FC<PropsFilter> = ({
+  todosInitial,
+  todosToRender,
+}) => {
+  const [initialComplete, setInitialComplete] = useState(todosInitial);
+  const [initialCompleteOriginal, setInitialCompleteOriginale] =
+    useState(todosInitial);
+
+  const [initialInput, setInitialInput] = useState(todosInitial);
+  const [initialInputOriginal, setInitialInputOriginale] =
+    useState(todosInitial);
+
+  const [inputValue, setInputValue] = useState('');
+
+  useEffect(() => {
```

Comment:
you don't need all these states here, just create two state(one for filterValue, and one for the query), move them to the App.tsx, and pass here as props the values of those states and setState functions

so this component will just change the state, you will filter the todos in the App